bundles/framework/layerselector2/view/FilterButtons.js

```diff
@@ -0,0 +1,180 @@
+Oskari.clazz.define("Oskari.layerselector2.view.FilterButtons",
+    function ( element ) {
+        var me = this;
+        this.mapLayerService = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
+        this.layerlistService = Oskari.getSandbox().getService('Oskari.mapframework.service.LayerlistService');
```

Comment:
You should pass layerlistService as parameter so the component can function without fetching an external reference and it's more obvious that it's a requirement for using the component.